api/handler/v1/project.go

```diff
@@ -2,25 +2,160 @@ package v1
 
 import (
 	"context"
+	"errors"
+	"strings"
+
+	grpczap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+
+	"github.com/odpf/shield/internal/project"
+	"github.com/odpf/shield/model"
 
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
 
 	shieldv1 "go.buf.build/odpf/gw/odpf/proton/odpf/shield/v1"
 )
 
+var grpcProjectNotFoundErr = status.Errorf(codes.NotFound, "project doesn't exist")
+
+type ProjectService interface {
+	Get(ctx context.Context, id string) (model.Project, error)
+	Create(ctx context.Context, project model.Project) (model.Project, error)
+	List(ctx context.Context) ([]model.Project, error)
+	Update(ctx context.Context, toUpdate model.Project) (model.Project, error)
+}
+
 func (v Dep) ListProjects(ctx context.Context, request *shieldv1.ListProjectsRequest) (*shieldv1.ListProjectsResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method not implemented")
+	logger := grpczap.Extract(ctx)
+	var projects []*shieldv1.Project
+
+	projectList, err := v.ProjectService.List(ctx)
+	if err != nil {
+		logger.Error(err.Error())
+		return nil, grpcInternalServerError
+	}
+
+	for _, v := range projectList {
+		projectPB, err := transformProjectToPB(v)
+		if err != nil {
+			logger.Error(err.Error())
+			return nil, grpcInternalServerError
+		}
+
+		projects = append(projects, &projectPB)
+	}
+
+	return &shieldv1.ListProjectsResponse{Projects: projects}, nil
 }
 
 func (v Dep) CreateProject(ctx context.Context, request *shieldv1.CreateProjectRequest) (*shieldv1.CreateProjectResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method not implemented")
+	logger := grpczap.Extract(ctx)
+	metaDataMap, err := mapOfStringValues(request.GetBody().Metadata.AsMap())
+	if err != nil {
+		logger.Error(err.Error())
+		return nil, grpcBadBodyError
+	}
+
+	slug := request.GetBody().Slug
+	if strings.TrimSpace(slug) == "" {
+		slug = generateSlug(request.GetBody().Name)
+	}
+
+	newProject, err := v.ProjectService.Create(ctx, model.Project{
+		Name:     request.GetBody().Name,
+		Slug:     slug,
+		Metadata: metaDataMap,
+		//Organization: org.Organization{Id: "ACCEPT"},
```

Comment:
Once odpf/proton#56 is merged, this would be updated

Reply:
Approved and merged.